tests/app/notifications/test_receive_notification.py

```diff
@@ -5,35 +5,89 @@
     strip_leading_forty_four
 )
 
-from app.models import InboundSms
+from app.models import InboundSms, EMAIL_TYPE, SMS_TYPE, INBOUND_SMS_TYPE
 from tests.app.db import create_service
+from tests.app.conftest import sample_service
 
 
-def test_receive_notification_returns_received_to_mmg(client, sample_service, mocker):
+def test_receive_notification_returns_received_to_mmg(client, mocker, sample_service_full_permissions):
     mocked = mocker.patch("app.notifications.receive_notifications.tasks.send_inbound_sms_to_service.apply_async")
-    data = {"ID": "1234",
+    data = {
+        "ID": "1234",
+        "MSISDN": "447700900855",
+        "Message": "Some message to notify",
+        "Trigger": "Trigger?",
+        "Number": "testing",
+        "Channel": "SMS",
+        "DateRecieved": "2012-06-27 12:33:00"
+    }
+    response = client.post(path='/notifications/sms/receive/mmg',
+                           data=json.dumps(data),
+                           headers=[('Content-Type', 'application/json')])
+
+    assert response.status_code == 200
+    assert response.get_data(as_text=True) == 'RECEIVED'
+    inbound_sms_id = InboundSms.query.all()[0].id
+    mocked.assert_called_once_with(
+        [str(inbound_sms_id), str(sample_service_full_permissions.id)], queue="notify-internal-tasks")
+
+
+@pytest.mark.parametrize('provider,headers,data,expected_response', [
+    (
+        'mmg',
+        [('Content-Type', 'application/json')],
+        json.dumps({
+            "ID": "1234",
             "MSISDN": "447700900855",
             "Message": "Some message to notify",
             "Trigger": "Trigger?",
             "Number": "testing",
             "Channel": "SMS",
             "DateRecieved": "2012-06-27 12:33:00"
-            }
-    response = client.post(path='/notifications/sms/receive/mmg',
-                           data=json.dumps(data),
-                           headers=[('Content-Type', 'application/json')])
+        }),
+        'RECEIVED'
+    ),
+    (
+        'firetext',
+        None,
+        {
+            "Message": "Some message to notify",
+            "source": "Source",
+            "time": "2012-06-27 12:33:00",
+            "destination": "447700900855"
+        },
+        '{\n  "status": "ok"\n}'
+    ),
+])
+@pytest.mark.parametrize('permissions', [
+    ([SMS_TYPE]),
+    ([INBOUND_SMS_TYPE]),
+])
+def test_receive_notification_without_permissions_does_not_create_inbound(
+        client, mocker, notify_db, notify_db_session, permissions, provider, headers, data, expected_response):
+    service = sample_service(notify_db, notify_db_session, permissions=permissions)
+    mocker.patch("app.notifications.receive_notifications.dao_fetch_services_by_sms_sender",
+                 return_value=[service])
+    mocked = mocker.patch("app.notifications.receive_notifications.tasks.send_inbound_sms_to_service.apply_async")
+    mocked_logger = mocker.patch("flask.current_app.logger.error")
```

Comment:
Do you have to mock a logger. Just thinking it's a bit nasty if we have to do this in any test where  code will have `current_app.logger.error`.